cloudinit/sources/DataSourceOpenStack.py

```diff
@@ -16,8 +16,11 @@
 
 LOG = logging.getLogger(__name__)
 
+
 # Various defaults/constants...
-DEF_MD_URLS = ["http://[fe80::a9fe:a9fe]", "http://169.254.169.254"]
+DEF_MD_URLS = [
+    "http://[fe80::a9fe:a9fe%{iface}]".format(iface=net.find_fallback_nic()),
```

Comment:
We don't want to call this function outside of a function definition. This would call `net.find_fallback_nic()` on all datasources (including non-openstack).

Reply:
@holmanb do you like this better? I moved the addresses to the only function they are used (and likely mangled).

Reply:
> @holmanb do you like it better when I move the addresses to the only function they are used (and likely mangled)?

Yes. This isn't referenced externally so that should be fine.

> Should / could I also use self.distro.fallback_interface ?

Please do, that would be preferred.